test/create_zarr_optimus.py

```diff
@@ -0,0 +1,307 @@
+
+    # ignore the first line with the cell metric names in text
+    cell_metric_values = []
+    for row in cell_metrics[1:]:
+        row_values = []
+        for value_string in row[1:]:
+            # some of the standard deviation values do not exist for one reads matches
+            try:
+                value = np.float32(value_string)
+            except ValueError:
+                value = np.nan
+            row_values.append(value)
+        cell_metric_values.append(row_values)
+
+    # Gene metric values, the row and column sizes
+    ncols = len(cell_metrics[0][1:])
+    nrows = len(cell_metrics[1:])
+
+    # now insert the dataset that has the numeric values for the cell qc metrics
+    data_group.create_dataset(
+        "qc_cell_values",
+        shape=(nrows, ncols),
+        compressor=COMPRESSOR,
+        dtype=np.float32,
+        chunks=(nrows, ncols),
+        data=cell_metric_values)
+
+
+def add_expression_counts(data_group, args):
+    """Converts  the count matrix from the Optimus pipeline to zarr file
+    Args:
+        data_group (zarr.hierarchy.Group): datagroup object for the zarr
+        args (argparse.Namespace): input arguments for the run
+    """
+    # read the cell ids and adds into the cell_barcodes dataset
+    barcodes = np.load(args.cell_ids)
+
+    if len(barcodes):
+        data_group.create_dataset(
+            "cell_id",
+            shape=(1, len(barcodes)),
+            compressor=COMPRESSOR,
+            dtype='<U40',
+            chunks=(1, len(barcodes)),
+            data=[list(barcodes)])
+
+    # read the gene ids  and adds into the gene_ids dataset
+    gene_ids = np.load(args.gene_ids)
+
+    if len(gene_ids):
+        data_group.create_dataset(
+            "gene_id",
+            shape=(1, len(gene_ids)),
+            compressor=COMPRESSOR,
+            dtype='<U40',
+            chunks=(1, len(gene_ids)),
+            data=[list(gene_ids)])
+
+    # read .npz file expression counts and add it to the expression_counts dataset
+    exp_counts = np.load(args.count_matrix)
+
+    # now convert it back to a csr_matrix object
+    csr_exp_counts = sparse.csr_matrix((exp_counts['data'],
+                                        exp_counts['indices'],
+                                        exp_counts['indptr']),
+                                       shape=exp_counts['shape'])
+
+    # now create a dataset of zeros with the same dimensions as the expression count matrix
+    exp_counts_group = data_group.zeros('expression',
+                                        compressor=COMPRESSOR,
+                                        shape=csr_exp_counts.shape,
+                                        chunks=(CHUNK_ROW_SIZE, csr_exp_counts.shape[1]),
+                                        dtype=np.float32)
+
+    # load a chunks from the expression count matrix and update the corresponding chunk in expression matrix in zar
+    for i in range(0, csr_exp_counts.shape[0], CHUNK_ROW_SIZE):
```

Comment:
nit: Since it uses `csr_exp_counts.shape[0]` in a lot a places, for a better readability, could we define a temp variable with a good name to store it? e.g.
```python
xxx_num_row, xxx_num_col = csr_exp_counts.shape
```